### PR TITLE
Verify Chain User Credentials Upon Login

### PR DESCRIPTION
**Pull request by AdrianAndersen**
This PR adds functionality to verify the chain user credentials users submit.

Depends on: https://github.com/mathiazom/rezervo-web/pull/102

### CODE REVIEW

File: rezervo/providers/ibooking/provider.py [hunk 1]
```diff
@@ -313,3 +313,15 @@ async def find_public_ibooking_class(
...
+
+    async def verify_authentication(self, credentials: ChainUserCredentials) -> bool:
+        async with ClientSession() as session:
+            auth_session = (
+                await authenticate_session(
+                    session, credentials.username, credentials.password
+                ),
+            )
+            return not isinstance(
+                auth_session[0],
```
Comment (AdrianAndersen) [inline rezervo/providers/ibooking/provider.py:325]: For some reason, this is a tuple, not a Union type in runtime

File: rezervo/api/chain_config.py [hunk 2]
```diff
@@ -50,13 +51,15 @@ def put_chain_user_creds(
...
+    if not await get_chain(chain_identifier).verify_authentication(chain_user_creds):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN)
     updated_config = crud.upsert_chain_user_creds(
         db, db_user.id, chain_identifier, chain_user_creds
     )
     if updated_config is None:
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND)
     background_tasks.add_task(refresh_cron, db_user.id, [chain_identifier])
-    return updated_config
+    return {"username": updated_config.username}
```
Comment (AdrianAndersen) [inline rezervo/api/chain_config.py:62]: This line currently throws an error in prod, that we just silently ignore. Reason being is that updated_config: ChainUser (username and password) is not compatible to the response_model: ChainUserProfile (only username)

Comment (mathiazom) [reply rezervo/api/chain_config.py:62]: Just saw this error in the prod logs, nice catch! Could maybe return an actual ChainUserProfile model with the username from `updated_config`, just for clarity?

Comment (AdrianAndersen) [reply rezervo/api/chain_config.py]: Ah yes updated in https://github.com/mathiazom/rezervo/pull/35/commits/d38fc5749ebcc9e2aa977e790a78271f981a3247

File: rezervo/providers/ibooking/provider.py (hunk 1 above)
Comment (mathiazom) [reply rezervo/providers/ibooking/provider.py:325]: The extra comma in line 322 makes it a [single element tuple](https://docs.python.org/3/tutorial/datastructures.html#tuples-and-sequences) :upside_down_face: 
```
auth_session = (
    (...),
)
```

**Review by mathiazom — APPROVED** [review]

**Pull request merged**